homeassistant/components/konnected.py

```diff
@@ -119,6 +126,33 @@ def device_discovered(service, info):
                             " but not specified in configuration.yaml",
                             discovered.device_id)
 
+    def device_discovered(service, info):
+        """Call when a Konnected device has been discovered."""
+        host = info.get(CONF_HOST)
+        port = info.get(CONF_PORT)
+        setup_device(host, port)
+
+    async def manual_discovery(event):
+        """Init devices on the network with manually assigned addresses."""
+        specified = [dev for dev in cfg.get(CONF_DEVICES) if
+                     dev.get(CONF_HOST) and dev.get(CONF_PORT)]
+
+        while specified:
+            for dev in specified:
+                _LOGGER.debug("Discovering Konnected device %s at %s:%s",
+                              dev.get(CONF_ID),
+                              dev.get(CONF_HOST),
+                              dev.get(CONF_PORT))
+                try:
+                    await hass.async_add_job(setup_device,
```

Comment:
Use `hass.async_add_executor_job`.

Reply:
If we're setting up different devices here, we can probably make this more efficient and try to set them all up concurrently using the asyncio machinery, eg using `asyncio.wait`. It gets a little tricky with the exception handling and additional sleep though. I'm not sure what's the best approach.